src/checkmatelib/client.py

```diff
@@ -41,10 +49,17 @@ def check_url(self, url, allow_all=False):
 
         params = {"url": url}
 
+        auth = None
+
         if allow_all:
             params["allow_all"] = True
 
-        response = requests.get(self._host + "/api/check", params=params, timeout=1)
+        if self.api_key:
+            auth = (self.api_key,)
+
+        response = requests.get(
+            self._host + "/api/check", params=params, timeout=1, auth=auth
+        )
```

Comment:
Could the setting of `auth` be moved into one place. For example:

```python
        auth = None

        if self.api_key:
            auth = (self.api_key,)

        response = requests.get(
            self._host + "/api/check", params=params, timeout=1, auth=auth
        )
```

or:

```python
        auth = (self.api_key,) if self.api_key else None

        response = requests.get(
            self._host + "/api/check", params=params, timeout=1, auth=auth
        )
```

or even:

```python
        response = requests.get(
            self._host + "/api/check",
            params=params,
            timeout=1,
            auth=(self.api_key,) if self.api_key else None,
        )
```

Reply:
Good point, went for the last one :+1: